src/relayclient/RelayClient.ts

```diff
@@ -62,33 +73,25 @@ export interface RelayingResult {
 export class RelayClient {
   readonly emitter = new EventEmitter()
   config!: GSNConfig
-  private readonly partialConfig?: { provider: HttpProvider, configOverride: Partial<GSNConfig>, overrideDependencies?: Partial<GSNDependencies> }
-  private httpClient!: HttpClient
-  protected contractInteractor!: ContractInteractor
-  protected knownRelaysManager!: KnownRelaysManager
-  private asyncApprovalData!: AsyncDataCallback
-  private asyncPaymasterData!: AsyncDataCallback
-  private transactionValidator!: RelayedTransactionValidator
-  private pingFilter!: PingFilter
-
-  public accountManager!: AccountManager
+  dependencies!: GSNDependencies
+  private readonly rawConstructorInput: GSNUnresolvedConstructorInput
+
   private initialized = false
   logger!: LoggerInterface
+  initializingPromise?: Promise<void>
 
-  /**
-   * create a RelayClient library object, to force contracts to go through a relay.
-   */
   constructor (
-    provider: HttpProvider,
-    configOverride: Partial<GSNConfig>,
-    overrideDependencies?: Partial<GSNDependencies>
+    rawConstructorInput: GSNUnresolvedConstructorInput
   ) {
-    this.partialConfig = { provider, configOverride, overrideDependencies }
-    this.logger = console
+    // TODO: backwards-compatibility 102 - remove on next version bump
+    if (arguments[0] == null || arguments[0].send != null || arguments[2] != null) {
+      throw new Error('Sorry, but the constructor parameters of the RelayClient class have changed. See "GSNUnresolvedConstructorInput" interface for details.')
+    }
+    this.rawConstructorInput = rawConstructorInput
+    this.logger = rawConstructorInput.overrideDependencies?.logger ??
+      createClientLogger(rawConstructorInput.partialConfig?.loggerConfiguration ?? defaultLoggerConfiguration)
```

Comment:
you override configuration in "all or nothing" manner: unable to change a single field

Reply:
There is nothing in the defaultLoggerConfiguration worth overriding.